### DISCUSSION TITLE
Copilot Not Suggesting for Jupyter Notebooks (PyCharm)

### DISCUSSION THREAD

**Discussion by Vio-Eli** [community/community#29319] [Copilot Conversations]
Hi,
I'm running PyCharm 2022.2 professional, with copilot version 1.1.29.1869 stable (also happens on the nightly build too)

Copilot gives suggestions for regular python files, but errors out when I try to autocomplete in jupyter notebooks.

When I go to Tools > Github Copilot > Open Github Copilot, it just shows an error
![image](https://user-images.githubusercontent.com/56001107/184457500-06d153b5-a348-4ab2-9e32-5b99518c7faf.png)

I added `#com.github.copilot` to the debug log configs, and here is a snippet (I can post the rest if needed, its just over 512kb so I can't pastebin it) of the resulting log:

```2022-08-12 16:31:31,675 [1185448]   WARN - #c.g.c.l.a.l.VirtualFileUri - Unable to parse as compliant URI, using fallback: notebook://F:/Code/Manga/CRAFT_Data_Gen/Manga_109s_Processing/Manga109s_released_2021_12_30/presentation.ipynb
2022-08-12 16:31:31,677 [1185450]   FINE - #c.g.c.l.a.CopilotAgentProcessServiceImpl - Registering result for request ID: 35, command: getPanelCompletions
2022-08-12 16:31:31,678 [1185451]   FINE - #c.g.c.l.a.v.VSCodeJsonRpcParser - Found content-length: 485
2022-08-12 16:31:31,678 [1185451]   FINE - #c.g.c.l.a.v.DefaultJsonRpcMessageHandler - received JSON-RPC error. id: 35, error: Couldn't find document for uri: notebook://f:/Code/Manga/CRAFT_Data_Gen/Manga_109s_Processing/Manga109s_released_2021_12_30/presentation.ipynb, known document uris are: file:///f%3A/Code/Manga/CRAFT_Data_Gen/Manga_109s_Processing/Manga_109s_processor.py, file:///f%3A/Code/Manga/CRAFT_Data_Gen/Manga_109s_Processing/Manga109s_released_2021_12_30/presentation.ipynb, request id: 35, error code: -32602
2022-08-12 16:31:31,679 [1185452]   FINE - #c.g.c.t.OpenCopilotHandler - Error fetching solutions: Error processing command getPanelCompletions, request id: 35, error code: -32602
```

Let me know if there is anything I can do to solve this! Thank you!

**Comment by jakubLangr** [comment]
Same issue, but I have VS Code.

**Comment by EthanHaque** [comment]
Same copilot issue with both pycharm 2022.2 and VS Code.

**Comment by Nicistarful** [comment]
Here to bump the thread, issue still remains on Pycharm 2022.3 and apparently 2022.2, the current solution is to use the version 2022.1.4, which isn't a long-term solution though.

**Comment by egg3000** [comment]
Same problem for me, am hoping for a quick bug fix by the Copilot team.

**Comment by aseedb** [comment]
same issue, hoping for a fix

**Comment by hjysmi** [comment]
same issue with below build.

PyCharm 2022.3 (Professional Edition)
copilot: 1.1.38.2229

**Comment by wang-ye** [comment]
same issue with pycharm 2022.2, copilot 1.1.38.2229

**Comment by agbragin** [comment]
Hi everyone, my name is Anton Bragin and I'm responsible for Jupyter notebook support in DataSpell and PyCharm. 

I would like to propose a fix for the issue which, I believe, should be done on [GitHub Copilot plugin](https://plugins.jetbrains.com/plugin/17718-github-copilot) side. 

Dear plugin authors, please contact me by anton.bragin at jetbrains.com, so we could discuss the fix.

**Reply by hstaudacher** [reply]
Hey @agbragin, I just dropped you a mail. Let me know if it has not reached you.

**Reply by 0xBaconPancakes** [reply]
Hey, any info about possible fix?

**Reply by kirlosik** [reply]
upup

**Reply by hstaudacher** [reply]
The latest [nightly build](https://plugins.jetbrains.com/plugin/17718-github-copilot/versions/nightly/276237) should have the error fixed. Please note, the completions copilot will provide will not be optimal due to the missing support for notebooks. We are in close contact with Anton from Jetbrains regarding a notebook API. When there will be a notebook API we can consume we will improve support for notebooks.

**Reply by galsalo** [reply]
Does this work for anyone? not for me...

**Reply by hstaudacher** [reply]
@galsalo I just checked again and it's working for me. Do you run the latest nightly? I have posted instructions how to do so here: https://youtrack.jetbrains.com/issue/DS-3756/Github-Copilot-does-not-work-in-DataSpell-with-Jupyter-notebooks#focus=Comments-27-6781098.0-0

**Reply by galsalo** [reply]
@hstaudacher Yeah I run the latest nightly, it works but the copilot suggestions made no sense. I then read your comment that we'll need to wait for a notebook api...(even though there was a period it worked perfectly, until it stopped working)

**Reply by hstaudacher** [reply]
The recent problems were due to the new notebook editor. If you edit your notebook in the text editor (e.g. by adding *.ipynb to text fies in the File Types settings) you will get normal inline completions with copilot considering the sorrounding notebook. Well, I would prefer a great support for notebooks in the new editor 😄

**Reply by AleksandrovichK** [reply]
Hey @hstaudacher is there a chance something’s updated?

**Reply by hstaudacher** [reply]
@AleksandrovichK check the comments from here: https://youtrack.jetbrains.com/issue/DS-3756/Github-Copilot-does-not-work-in-DataSpell-with-Jupyter-notebooks#focus=Comments-27-6813688.0-0
